crates/once-map/src/lib.rs

```diff
@@ -46,41 +47,61 @@ impl<K: Eq + Hash, V: Clone, H: BuildHasher + Clone> OnceMap<K, V, H> {
     ///
     /// Will hang if [`OnceMap::done`] isn't called for this key.
     pub async fn wait(&self, key: &K) -> Option<V> {
-        let entry = self.items.get(key)?;
```

Comment:
I'm trying to reason through how a deadlock could happen here. My understanding is that `entry` here is actually a [`dashmap::mapref::one::Ref`](https://docs.rs/dashmap/latest/dashmap/mapref/one/struct.Ref.html), and that in turn holds a lock while it's alive. (The `dashmap` docs are woefully incomplete, but it's what the implementation suggests.) If that's true, then once a `get` happens, then it shouldn't be possible for a `done` call to insert anything before the case analysis below, right?

Oh... wait... OK. I now see the `drop(entry)` below just before the `notify.notified().await`. So the deadlock is that between `drop(entry)` and `notify.notified().await`, a `done` call inserts a filled entry for the given key and notifies waiters _before_ `notify.notified()` has a chance to register the waiter.